### DISCUSSION TITLE
Update ref element of popper without making a new instance

### DISCUSSION THREAD

**Discussion by theo-staizen** [floating-ui/floating-ui#1214] [Q&A] [answered]
Is this the "correct" way to update the ref element, because this feels like i'm touching internals:

```js
const popper =  createPopper(ref1, popEl, {});

setTimeout(() => {
  popper.state.elements.reference = ref2;
  popper.update();
}, 1000);
```

Full example: https://codepen.io/dogoku/pen/vYXaxML

**Comment by FezVrasta** [comment, marked as answer]
Changing the reference or popper element is not supported, you should destroy the instance and make a new one. It's a very quick operation.

**Reply by theo-staizen** [reply]
So then the above example would become this?

```js
let popper =  createPopper(ref1, popEl, {});

setTimeout(() => {
  popper.destroy();
  popper = createPopper(ref2, popEl, {});
}, 1000);
```

Thanks for the quick reply, that seems to be working

**Reply by FezVrasta** [reply]
Yes that's the proper way to do it.
